apps/passport-client/src/service-worker.ts

```diff
@@ -1,65 +1,307 @@
 // this file is loaded as a service worker
 
+// Hack to make TypeScript aware of the ServiceWorkerGlobalScope type.
+// We can't redeclare `self`, but `swSelf` can be used with the right type.
+/// <reference lib="webworker" />
+const swSelf = self as unknown as ServiceWorkerGlobalScope;
+
+// Global flag to enable or disable service worker quickly.
+export const SERVICE_WORKER_ENABLED =
+  true || process.env.NODE_ENV !== "development"; // TODO(artwyman): cleanup before checkin
+
 /**
- * These files are cleared from the cache every time a new version
- * of the service worker is installed, because a new version of
- * the service worker is triggered by a change in the service worker
- * source code, which happens each time production deploys.
+ * Ephemeral cache is for resources which may change frequently, where we
+ * want to optimize for freshness, but allow caching for offline.  This cache
+ * is also updated dynamically as resources are fetched.  Cache name includes
+ * deployed code version, which protects against using
+ * cached pages which came from a different server deploy.
  */
-const impermanentCache = [
-  // "/",
-  // "/favicon.ico",
-  // "/index.html",
-  // "/global-pcdpass.css",
-  // "/global-zupass.css",
-  // "/js/index.js",
+const EPHEMERAL_CACHE_NAME = `v1-${process.env.SW_ID}`;
+const EPHEMERAL_CACHE_RESOURCES = [
+  "/",
+  "/index.html",
+  "/global-zupass.css",
+  "/js/index.js"
 ];
 
-async function addResourcesToCache(resources: string[]): Promise<void> {
-  const cache = await caches.open("v1");
+/**
+ * Stable cache is for resources which are expected not to change frequently,
+ * where we want to optimize for speed and limit bandwidth at the cost of
+ * freshness.  This cache never has additional resources added.
+ * These resources are updated only when a new service worker is installed.
+ * since htat update isn't guaranteed to succeed, this list shouldn't contain
+ * anything which could cause compatibility problems if it isn't updated.
+ */
+const STABLE_CACHE_NAME = "v1-stable";
```

Comment:
may be worth splitting out the version `v1` to another constant with documentation regarding when to change it.